tests/selenium/spec/signup-spec.js

```diff
@@ -0,0 +1,57 @@
+const SignUpPage = require("../framework/page-objects/SignUpPage");
+const util = require('../framework/shared/util');
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+
+describe('Sign up page check spec', () => {
+  const signUpPage = new SignUpPage();
+
+  beforeEach(util.itHelper(async () => {
+    await signUpPage.navigate();
+  }));
+
+  it('shows all the fields and button', util.itHelper(async () => {
+    expect(await signUpPage.isEmailFieldPresent(), 'expects email field to be present').to.equal(true);
+
+    expect(await signUpPage.isFirstNameFieldPresent(), 'expects first name field to be present').to.equal(true);
+
+    expect(await signUpPage.isLastNameFieldPresent(), 'expects last name field to be present').to.equal(true);
+
+    expect(await signUpPage.isCountryFieldPresent(), 'expects country field to be present').to.equal(true);
+
+    expect(await signUpPage.isSignUpByEmailButtonPresent(), 'expects signup button to be present').to.equal(true);
+
+    expect(await signUpPage.isSignUpByGithubButtonPresent(), 'expects signup with github button to be present').to.equal(true);
+
+    expect(await signUpPage.isSignUpByGoogleButtonPresent(), 'expects signup with google button to be present').to.equal(true);
+
+    expect(await signUpPage.isSignInLinkPresent(), 'expects sign in link to be present').to.equal(true);
+
+  }));
+
+  describe('button clicks check', () => {
+
+    it('sign up with github button click takes user to github', util.itHelper(async () => {
+      await signUpPage.getGithubSignUpButtonElement().click();
+      expect(await signUpPage.getCurrentURL())
+        .to.contain("github.com/login");
+    }));
+
+    it('sign up with google button click takes user to google', util.itHelper(async () => {
+      await signUpPage.getGoogleSignUpButtonElement().click();
+      expect(await signUpPage.getCurrentURL())
+        .to.contain("accounts.google.com");
+    }));
+
+    it('sign in link click takes user to login page', util.itHelper(async () => {
+      await signUpPage.getSignInLinkElement().click();
+      expect(await signUpPage.getCurrentURL())
+        .to.contain("login");
+    }));
+
```

Comment:
We can add another functional test here to see if the text fields especially email fields and country field works as expected and in the end checking if the I agree check box shows up and possibly if clicking sign up takes us to a thank you page.

Reply:
I have added agree check box and buttons text tests. for actually clicking sign up, I will be creating separate jira as that includes interaction with captcha and might include some investigation